### PR TITLE
Blog file was added

### PR DESCRIPTION
**Pull request by jacobsakran**
Report for Benchmark was added

### CODE REVIEW

**Comment by eranra** [comment]
I am looking at the graphs .... sometimes 4 ingesters are blue and sometimes 2 ingesters are blue ... can you change the colors so that the graphs are consistent

File: blog/report.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# Benchmark Report
+
+## Goal
+We want to give the user the opportunity to know the minimum size of OCP cluster that can stand for his daily logs written and read from Loki. Our goal is to create a table that specifies for each "T-shirt" size the average and max daily read and write capabilities of Loki.
```
Comment (eranra) [inline blog/report.md:4]: s/minimum size of OCP cluster/recommended LokiStack configuration

explain --> S/XXX/YYY stands for replacing XXX with YYY

Comment (eranra) [reply blog/report.md:4]: s/OCP/OpenShift

File: blog/report.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+## Environment
+We ran that on OpenShift running on AWS clusters with configuration:
+
+	apiVersion: v1
+	baseDomain: devcluster.openshift.com
```
Comment (eranra) [inline blog/report.md:10]: drop `baseDomain` and `apiVersion` this is not needed

File: blog/report.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# Benchmark Report
+
+## Goal
+We want to give the user the opportunity to know the minimum size of OCP cluster that can stand for his daily logs written and read from Loki. Our goal is to create a table that specifies for each "T-shirt" size the average and max daily read and write capabilities of Loki.
+
+## Environment
+We ran that on OpenShift running on AWS clusters with configuration:
```
Comment (eranra) [inline blog/report.md:7]: s/that/the benchmark

File: blog/report.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+			iops: 8000
+	replicas: 6
+	controlPlane:
+	architecture: amd64
+	hyperthreading: Enabled
+	name: master
+	platform: {}
+	replicas: 3
+
+Running with 64 vCPUs and 256 Memory (GiB)
```
Comment (eranra) [inline blog/report.md:30]: Add: In this configuration each node has

File: blog/report.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+
+## The Proccess Of Creating The Table
+The benchmark tool helps us observe the data (such as CPU, Memory, etc...) of Loki while sending a different amount of logs each test, the logs are sent by a logger that is deployed when the benchmark test is started (the implementation can be seen at https://github.com/ViaQ/cluster-logging-load-client).
```
Comment (eranra) [inline blog/report.md:34]: s/helps us observe the data/records operational metrics
s/each test,/in each test.

File: blog/report.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+We have been observing mainly the CPU (in MiliCore) and Memory (in GB) as a function of the GiPD (Log load in GB per Day) to see how they change, we ran the benchmark test with different values of GiPD and watched over the CPU and Memory and how the number of Ingesters may affect these values in Loki. First, we ran Loki with 2 Ingesters and then we ran it with 4 Ingesters.
```
Comment (eranra) [inline blog/report.md:36]: s/, we/. We

File: blog/report.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+Then a high amount of logs has been sent to Loki by running different values between 2250GB and 6000GB per day (starting with 2250Gb then 2500GB ... and so on until 6000GB). Sending this amount of GB to Loki with 2 Ingesters made the GiPD go too high compared to what has been sent, but the GiPD in Loki with 4 Ingesters was pretty much normal.
+
```
Comment (eranra) [inline blog/report.md:39]: @jacobsakran this looks very good to me. I added comments above for you to fix 

@Red-GV can you review?

File: blog/report.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+
```
Comment (eranra) [inline blog/report.md:40]: @jacobsakran I am missing two more sections here ... I think we need:

A section called "conclusions" with:
1. From the results we can learn that .... 
2. The information shows clearly that, and the information is not clear on ... 
etc. 

Also, missing is a "TODO/Next" section that talks about what was done and what is still missing. For example, read tests that we did not do at all ... and for example write tests with different values or different scale of different LokiStack components

File: blog/report.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+## The Commands That Have Been Used:
+
+Getting the GiPD:
```
Comment (eranra) [inline blog/report.md:43]: s/Getting/Observing (or monitoring)

**Review by eranra — COMMENTED** [review]
Looks very good @jacobsakran thanks for pulling that together ... please work on the comments and we will work to try and merge this in.

Also, I am not sure that this needs to go into **blog** folder.In any event please add a link to the README to  link into this blog

**Review by eranra — COMMENTED** [review]
Also, If you can squash those15 commits into a single one

**Review by Red-GV — CHANGES_REQUESTED** [review]
@eranra @jacobsakran I don't think that the repository is the right medium for this report. This does not add any capabilities nor information for operating the tool.

**Comment by eranra** [comment]
Hi @Red-GV  ... I would not mind putting that anywhere else (ideas??)  but I think that having a report (I would not call that blog but report) with results showing the value of the benchmark tool and what it can do for specific environments like OpenShift as part of the repo makes a lot of sense, why not??

**Comment by eranra** [comment]
@Red-GV in any event, if you can review the text + graphs so we can get this ready here or somewhere else

**Comment by Red-GV** [comment]
@eranra I would think that a better place for reports showing potential performance would be better on websites (like Medium). 

Keeping them in a repository is not very beneficial because this report is something very "specific". For instance, the justification for merging in this would be the same as letting in any report with some statistics for any type of configurations of Loki. That can be potentially bloating.

Mainly, the the repository is for operating the tool. Not for keeping track of the reports or compilations of them.

**Comment by eranra** [comment]
@Red-GV Sure ... we can do that ... so let's use this PR to agree on text and content and once we are OK with that @jacobsakran  will be able to upload to `medium`

File: blog/report.md (hunk 8 above)
Comment (eranra) [reply blog/report.md:38]: @jacobsakran can you add another section with TODO/Next that will talk about what is still missing in this evaluation

**Comment by eranra** [comment]
@jacobsakran one comment that I missed (probably) is that the colors of the graphs are misleading. Sometimes blue is 2 ingesters and sometimes it is brown. Can you fix it so that the same color will represent the same ingesters count

**Comment by eranra** [comment]
Adding a conclusion section is good. Something is wrong with the indent of the points. When I look on 1. and 2. the text under those points is shown strange.

Also, with regards to the text itself ... if you can talk about the entire spectrum and not on specific points in the graph. For example, say something similar to "we can see that when stressing Loki with log writes up to a level of 2000 GB of data per day, it is clear that with a configuration of 2 ingesters Loki handles the load with a reasonable amount of resources (1 core and 100GB of memory). We also see that load that is above that level is causing the 2 ingesters to consume a lot more memory and CPU, this is a good point to suggest to move into 4 ingesters."

**Comment by Red-GV** [comment]
Closing PR. As stated, this repository is not meant to act as an archive for generated reports.

**Pull request closed without merging**